cmd/admin-handlers-users.go

```diff
@@ -2453,11 +2457,16 @@ func (a adminAPIHandlers) ImportIAM(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func addExpirationToCondValues(exp *time.Time, condValues map[string][]string) {
+func addExpirationToCondValues(exp *time.Time, condValues map[string][]string) error {
 	if exp == nil {
-		return
+		return nil
+	}
+	dur := exp.Sub(time.Now())
```

Comment:
```suggestion
	dur := exp.Sub(time.Now())
```

check if `exp.IsZero()` is true or `exp.Equal(time.Unix(0,0))`  return `nil`

Reply:
Both would make the `if dur <= 0 {` below fail AFAICT, so it seems a bit superfluous.

Reply:
@klauspost though Expiration is a pointer, some clients a non nil expiration which is zero to mean I did not set any expiration in this request. I believe this is what Harsha meant. Fixed it